runtime/protocol/http-client-engines/http-client-engine-ktor/jvm/src/aws/smithy/kotlin/runtime/http/engine/ktor/KtorEngine.kt

```diff
@@ -38,6 +39,7 @@ class KtorEngine(val config: HttpClientEngineConfig) : HttpClientEngineBase("kto
     }
     private val logger = Logger.getLogger<KtorEngine>()
 
+    @OptIn(InternalAPI::class)
```

Comment:
FYI I was unable to get other variations of this to work.  Annotating at the class level or function level of `@InternalApi` throws an error saying that super needs the same annotation but super does not depend on ktor.  I found this to be a better than adding a dependency on ktor.

Reply:
Ktor's internal APIs are not stable and are subject to change. Presumably, the reason they're public is to allow sharing between subprojects within Ktor, not to allow 3P callers to reuse them. Shouldn't we replace these calls with either stable APIs or our own custom implementations?

Reply:
This is more than likely a bug in ktor. The [HttpRequestBuilder](https://github.com/ktorio/ktor/blob/main/ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt#L63) type is a pretty fundamental type used throughout the `HttpClient` and `HttpRequestPipeline/HttpResponsePipeline`.

Reply:
Indeed: https://youtrack.jetbrains.com/issue/KTOR-3001

They basically just said to optin and it will be removed in a future release.

Reply:
Ah, I missed that in my searching yesterday. Then I agree accepting opt-in seems like our only recourse for now.

Reply:
I'm fine with that and it looks like we won't have to wait long:

"We’re aiming to release 2.0 hopefully by October."
- https://blog.jetbrains.com/ktor/2021/08/10/an-update-on-our-2020-2021-roadmap/